data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCrac.java

```diff
@@ -481,14 +486,32 @@ public void synchronize(Network network) {
         }
         branchCnecs.values().forEach(cnec -> cnec.synchronize(network));
         rangeActions.values().forEach(rangeAction -> rangeAction.synchronize(network));
+        contingencies.values().forEach(contingency -> contingency.synchronize(network));
+        addXnodeContingenciesNetworkElements();
         networkDate = network.getCaseDate();
         isSynchronized = true;
     }
 
+    /**
+     * This method adds network elements from XnodeContingencies to the crac.
+     * For ComplexContingencies, they are added when the contingencies are added to the crac. This is not possible for
+     * XnodeContingencies since they have to be synchronized with the network first in order to find the network elements
+     * corresponding to the xnodes.
+     */
+    private void addXnodeContingenciesNetworkElements() {
+        contingencies.values().stream().filter(contingency -> contingency instanceof XnodeContingency)
+                .forEach(contingency -> contingency.getNetworkElements().forEach(networkElement -> {
+                    if (getNetworkElement(networkElement.getId()) == null) {
+                        addNetworkElement(networkElement);
+                    }
+                }));
+    }
+
```

Comment:
It should actually ideally be done directly by the synchronize method of the XnodeContingency. But it's not so easy to do with the current signature of the `synchronize()` method.

So i don't like it, but don' really have a better solution :p

Reply:
This was necessary when re-importing synced XnodeContingencies from a json file. Since we're not exporting sync information anymore, maybe we can remove this. hoping it works in all other cases. But I'm not really sure it will, since the XnodeContingency will reference network elements not present in the crac..

Reply:
do we keep it as a safety net ?

Reply:
Yes, I think it is safer to keep it